[oneTBB] Changing description for containers deduction guides (#355)
oneapi-src/oneAPI-spec: source/elements/oneTBB/source/containers/concurrent_unordered_multiset_cls/deduction_guides.rst

================

Where possible, constructors of ``concurrent_unordered_multiset`` support

> class template argument deduction (since C++17). Copy and move constructors (including constructors with explicit

Reviewer: ```suggestion
class template argument deduction (since C++17). Copy and move constructors, including constructors with an explicit
```

> ``allocator_type`` argument) provides implicitly generated deduction guides. In addition, the following explicit

Reviewer: ```suggestion
``allocator_type`` argument, provide implicitly generated deduction guides. In addition, the following explicit
```

    concurrent_unordered_multiset( std::initializer_list<T>,
                                   set_size_type,
                                   Allocator )
    -> concurrent_unordered_multiset<T,
                                     std::hash<T>,
                                     std::equal_to<T>,
                                     Allocator>;

    template <typename T,
              typename Allocator>
    concurrent_unordered_multiset( std::initializer_list<T>,
                                   Allocator )
    -> concurrent_unordered_multiset<T,
                                     std::hash<T>,
                                     std::equal_to<T>,
                                     Allocator>;

    template <typename T,
              typename Hash,
              typename Allocator>
    concurrent_unordered_multiset( std::initializer_list<T>,
                                   set_size_type,
                                   Hash,
                                   Allocator )
    -> concurrent_unordered_multiset<T,
                                     Hash,
                                     std::equal_to<T>,
                                     Allocator>;

> Where the type ``set_size_type`` refers to the ``size_type`` member type of the deduced ``concurrent_unordered_multiset``

Reviewer: ```suggestion
Where the ``set_size_type`` type refers to the ``size_type`` member type of the deduced ``concurrent_unordered_multiset``
```

and the type alias ``iterator_value_t`` is defined as follows:

.. code:: cpp

    template <typename InputIterator>
    using iterator_value_t = typename std::iterator_traits<InputIterator>::value_type;

> These deduction guides only participates in overload resolution if all of the following are ``true``:

Reviewer: ```suggestion
These deduction guides only participate in the overload resolution if all of the following are ``true``:
```

> * The type ``InputIterator`` meets the requirements of  ``InputIterator`` from the [input.iterators] ISO C++ Standard section.

Reviewer: ```suggestion
* The ``InputIterator`` type meets the ``InputIterator`` requirements described in the [input.iterators] section of the ISO C++ Standard.
```

> * The type ``Allocator`` meets the requirements of ``Allocator`` from the [allocator.requirements] ISO C++ Standard section.

Reviewer: ```suggestion
* The ``Allocator`` type meets the ``Allocator`` requirements described in the [allocator.requirements] section of the ISO C++ Standard.
```

> * The type ``Hash`` does not meet the requirements of ``Allocator``.

Reviewer: ```suggestion
* The ``Hash`` type does not meet the ``Allocator`` requirements.
```

> * The type ``KeyEqual`` does not meet the requirements of ``Allocator``.

Reviewer: ```suggestion
* The ``KeyEqual`` type does not meet the ``Allocator`` requirements.
```
